hibernate-core/src/main/java/org/hibernate/dialect/ColumnQualifierSupport.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.dialect;
+
+/**
+ * Indicates the level of qualifier support used by
+ * the dialect when referencing a column.
+ *
+ * @author Marco Belladelli
+ */
+public enum ColumnQualifierSupport {
+	/**
+	 * Qualify the column using the table expression,
+	 * ignoring a possible table alias.
+	 */
+	TABLE_EXPRESSION,
+
+	/**
+	 * Qualify the column using the table alias, whenever available,
+	 * and fallback to the table expression.
+	 */
+	TABLE_ALIAS,
```

Comment:
I have done this, but only when visiting delete and update statements. I think we still need to keep the `rendersTableReferenceAlias` method in `renderNamedTableReference` to render aliases for non UPDATE/DELETE clauses.

Reply:
Ok right, then let's re-model that method maybe to not `return false;` for `UPDATE`/`DELETE`, but instead return whether ` == DmlTargetColumnQualifierSupport.TABLE_ALIAS`

Reply:
To do this we would need access to the `Dialect`, which we don't have since the method is `static`. It's also `public`, so I didn't want to change the signature. Should I create another method (non-static or even 2 params) and deprecate this one?

Reply:
It's incubating, so no need to keep the old method.